cordformation/src/main/kotlin/net/corda/plugins/DockerImage.kt

```diff
@@ -0,0 +1,133 @@
+package net.corda.plugins
+
+import org.gradle.api.DefaultTask
+import com.spotify.docker.client.DefaultDockerClient
+import org.gradle.api.file.*
+import org.gradle.api.model.ObjectFactory
+import org.gradle.api.tasks.*
+import java.io.File
+import java.nio.file.Path
+import java.nio.file.Paths
+import javax.inject.Inject
+
+
+open class DockerImage  @Inject constructor(objects: ObjectFactory, layouts: ProjectLayout) : DefaultTask() {
+
+    private companion object{
+        //Currently a bug in gradle's projectFile that writes the file with a lower case 'd'
+        //using this name ensures that the docker plugin can find the file as expected.
+        private const val DOCKER_FILE_NAME = "dockerfile"
+    }
+
+    init {
+        description = "Creates a docker file and immediately builds that image to the local repository."
+        group = "CordaDockerDeployment"
+    }
+
+    @get:Input
+    internal var baseImage: String? = null
+
+    fun baseImage(baseImage: String) {
+        this.baseImage = baseImage
+    }
+
+    @get:Optional
+    @get:InputFile
+    internal var trustRootStoreFile: File? = null
+
+    fun trustRootStoreFile(trustRootStoreFile: File) {
+        this.trustRootStoreFile = trustRootStoreFile
+    }
+
+    @get:Optional
+    @get:Input
+    internal var buildImage: Boolean = true
+
+    fun buildImage(doBuild: Boolean) {
+        this.buildImage = doBuild
+    }
+
+    @get:Optional
+    @get:Input
+    internal var dockerImageTag: String? = null
+
+    fun dockerImageTag(tag: String) {
+        dockerImageTag = tag
+    }
+
+    private val _jars: ConfigurableFileCollection = project.files()
+
+    @get:InputFiles
+    @get:SkipWhenEmpty
+    val cordaJars: FileCollection get() = _jars
+
+    @Suppress("MemberVisibilityCanBePrivate")
+    fun setCordaJars(inputs: Any?) {
+        val files = inputs ?: return
+        _jars.setFrom(files)
+    }
+    fun cordaJars(inputs: Any?) = setCordaJars(inputs)
+
+    @get:OutputFiles
+    val dockerJars: FileCollection get() = project.files(cordaJars.map(::transferToDockerDir))
+
+    private fun transferToDockerDir(source: File) = outputDir.file(source.name)
```

Comment:
You are referring to the `outputDir` property before Kotlin has been able to assign it; I think you need to move both `dockerJars` and `transferToDockerDir` _below_ the `outputDir` property.

I also think you need to make this `dockerJars` property _lazy_:
```kotlin
@get:OutputFiles
val dockerJars: Property<FileCollection> = objects.property(FileCollection::class.java)
    .value(project.files(cordaJars.map(::transferToDockerDir)))
```
because the `outputDir` property is also lazy and you don't want Gradle to try and evaluate it until all configuration is complete.

Reply:
Reading a bit more of the OutputFiles annotation documentation I don't think this task really needs it and it seems to be slightly more hassle then it's worth to save the the task from rebuilding the dockerimage.